src/audio/igo_nr/igo_nr.c

```diff
@@ -43,46 +43,79 @@ DECLARE_SOF_RT_UUID("igo-nr", igo_nr_uuid,  0x696ae2bc, 0x2877, 0x11eb, 0xad, 0x
 
 DECLARE_TR_CTX(igo_nr_tr, SOF_UUID(igo_nr_uuid), LOG_LEVEL_INFO);
 
+static void igo_nr_lib_process(struct comp_data *cd)
+{
+	/* Pass through the active channel if
+	 * 1) It's not enabled, or
+	 * 2) hw parameter is not valid.
+	 */
+	if (!cd->process_enable[cd->config.active_channel_idx] || cd->invalid_param) {
```

Comment:
so we still deinterleave even when not active?

Reply:
Yes. Different buffer flow control is more likely to be error prone. Better to keep the same at expense of (tiny) efficiency sacrifice.